src/PKSim.Core/Services/BoxWhiskerChartsDataCreator.cs

```diff
@@ -64,37 +67,70 @@ private void setSeriesValues(CurveData<BoxWhiskerXValue, BoxWhiskerYValue> serie
 
          series.Add(xValue, yValue);
       }
-   }
 
-   internal class BoxWhiskerXValueComparer : IComparer<BoxWhiskerXValue>
-   {
-      private readonly IReadOnlyList<IPopulationAnalysisField> _columnFields;
-
-      public BoxWhiskerXValueComparer(IReadOnlyList<IPopulationAnalysisField> columnFields)
+      public override ChartData<BoxWhiskerXValue, BoxWhiskerYValue> CreateFor(PivotResult pivotResult)
       {
-         _columnFields = columnFields;
+         var chartData = base.CreateFor(pivotResult);
+         var boxWhiskerAnalysis = pivotResult.Analysis as PopulationBoxWhiskerAnalysis;
+         if (boxWhiskerAnalysis == null || chartData==null)
+            return chartData;
+
+         chartData.Panes.Each(p => updateIndividualIdForField(pivotResult.PopulationDataCollector, boxWhiskerAnalysis, p));
+
+         return chartData;
       }
 
-      public int Compare(BoxWhiskerXValue arg1, BoxWhiskerXValue arg2)
+      private void updateIndividualIdForField(IPopulationDataCollector populationDataCollector, PopulationBoxWhiskerAnalysis boxWhiskerAnalysis, PaneData<BoxWhiskerXValue, BoxWhiskerYValue> dataPane)
       {
-         if (!_columnFields.Any())
-            return 0;
+         var field = boxWhiskerAnalysis.FieldByName(dataPane.Id) as PopulationAnalysisNumericField;
+         if (field == null)
+            return;
 
-         if (arg1.Count != _columnFields.Count)
-            throw new ArgumentException(PKSimConstants.Error.InconsistentXValuesLength(arg1.Count, _columnFields.Count));
+         var fieldValues = field.GetValues(populationDataCollector).Where(x => x.IsValid()).ToFloatArray();
+         var sortedValues = fieldValues.OrderBy(x => x).ToList();
 
-         if (arg2.Count != _columnFields.Count)
-            throw new ArgumentException(PKSimConstants.Error.InconsistentXValuesLength(arg2.Count, _columnFields.Count));
+         dataPane.Curves.SelectMany(x => x.YValues).Each(y => updateIndexes(y, fieldValues, sortedValues));
+      }
 
-         int i;
-         for (i = 0; i < _columnFields.Count - 1; i++)
+      private void updateIndexes(BoxWhiskerYValue boxWhiskerYValue, float[] fieldValues, List<float> sortedValues)
+      {
+         boxWhiskerYValue.AllValues.Each(v =>
          {
-            int result = _columnFields[i].Compare(arg1[i], arg2[i]);
-            if (result != 0)
-               return result;
-         }
+            int index = sortedValues.BinarySearch(v);
+            v.IndividualId = Array.IndexOf(fieldValues, index >= 0 ? v : sortedValues[~index]);
```

Comment:
That's the actual retrieval of the IndividualId. I cannot do it in the aggregate function because we do not have access to the full dataset